### ISSUE TITLE
NVDA reads selected value as "empty"

### ISSUE THREAD

**Issue by Chrischaaan** [ng-select/ng-select#2478]
### Describe the bug

When a value is selected, NVDA reads the value of the Input as "empty

### Steps to Reproduce

1. Get NVDA and start it
2. go to [Stackblitz](https://stackblitz.com/edit/ng-select) and take one of the NG-SELECTs
3. Select a value
4. Focus the ng-select with tab --> value is read as "empty

### Please provide a link to a minimal reproduction of the bug. StackBlitz, CodePen or CodeSanBox

https://stackblitz.com/edit/ng-select

### Expected behaviour

The Value should be read by the screenreader. More correctly, the value should be set to the input. not a label-template...

### What version of the library you see this issue?

13.9.1

### What version of Angular are you using?

Angular 18

### Anything else?

The Value is not set to the Input correctly. if i write it to the input of the ng-select, its read correctly.
for example with "document.getElementById(...).value and make the label-temp hidden.
the value is only in the "ng-label-tmp" which is shown over the input.

at all, we use a control which is bound to the input and has the correct value - but this doesn't fix the problem.

**Comment by komsa-che1** [comment]
I have the same issue for months now. Need this for A11y purposes. Hoped, it will be fixed with a Version > 12. But doesn't work.

**Comment by pavankjadda** [comment]
@Chrischaaan @komsa-che1 how do I test this? Care to contrinute?

**Comment by pavankjadda** [comment]
> 1. NVDA

I thought, NVDA as stock. You meant https://www.nvaccess.org/download/?

**Comment by komsa-che1** [comment]
I have investigated the problem for a long time and tried various things. in my opinion, the problem is, that the input's value is set to '' after the searchTerm was cleared.

See line 51 in https://github.com/ng-select/ng-select/blob/master/src/ng-select/lib/ng-select.component.html

If i set the input-value to "selectedItems[0].label" (if available), then it works. but there must be a better Solution, cause this will not work with MultiSelect...

![image](https://github.com/user-attachments/assets/df8ba63d-7e0a-4de9-b269-a0f7366f55cf)

**Comment by Chrischaaan** [comment]
the isue is still tagged with "needs reproduce" @pavankjadda could you reproduce this behaviour?

**Comment by pavankjadda** [comment]
> the isue is still tagged with "needs reproduce" @pavankjadda could you reproduce this behaviour?

I haven't checked. If you can, could send a PR?

**Comment by Chrischaaan** [comment]
> > the isue is still tagged with "needs reproduce" @pavankjadda could you reproduce this behaviour?
> 
> I haven't checked. If you can, could send a PR?

Not in the next Weeks  i guess... Maybe after Christmas i can have a look.

**Comment by Chrischaaan** [comment]
> > > the isue is still tagged with "needs reproduce" @pavankjadda could you reproduce this behaviour?
> > 
> > 
> > I haven't checked. If you can, could send a PR?
> 
> Not in the next Weeks i guess... Maybe after Christmas i can have a look.

Currently i dont have time to fix this. At all, my Solution whould require, that the label-Template is removed. The Text has to be in the Input. The Label/MultiLabel-Template would cover the Text. This has issues to the Funktionality with Icons in the "value" and so on...

**Comment by komsa-che1** [comment]
Do you have an working StackBlitz Example for your fix? In my Project it is still read as "empty".

**Comment by Chrischaaan** [comment]
@pavankjadda Which exact behaviour does your PullRequest fix? The Content is still read as empty. As @komsa-che1 mentioned [here](https://github.com/ng-select/ng-select/issues/2478#issuecomment-2505426633), the input must have the value from the selected item(s).

**Comment by pavankjadda** [comment]
@komsa-che1 @Chrischaaan reopened this. I don't have windows machine, so can't test it really. Any suggestions?

**Comment by Chrischaaan** [comment]
Good morning,

is there any Screenreader available for your Maschine? does it work with this one?

**Comment by komsa-che1** [comment]
> Good morning,
> 
> is there any Screenreader available for your Maschine? does it work with this one?

Maybe this one @pavankjadda for Linux: https://help.gnome.org/users/orca/stable/index.html.en

The WIndows ScreenReader reads the value as "Leertaste" (== space key? for englisch users) instead of "Leer" (== empty in NVDA).
Maybe there is a Speec Program on your System preinstalled?

**Comment by Chrischaaan** [comment]
@pavankjadda 
i tried the following things like @komsa-che1 mentioned [here](https://github.com/ng-select/ng-select/issues/2478#issuecomment-2505426633)

![Image](https://github.com/user-attachments/assets/3a6d07b3-3443-484c-af73-549f5f60db5e)

ScreenReaders are fine with this, because the value is now in the input-value correctly.
But with this solution you can not use the "labelTemplate" and "multiLabelTemplate" anymore, because the values are shown twice in the field. One from the:

![Image](https://github.com/user-attachments/assets/3d9fbe11-bcc8-47d4-b4fc-14074b2819fe)

**Comment by bederuijter** [comment]
> "ariaLabel KindOfField selectedItem EMPTY".
>
> For Example, die Label of the Field is "Manufacturer" and you have selected an item with the Name "Porsche", then it reads "Manufacturer Combinations Porsche EMPTY"
> 
> it's only a part of the solution...

I've gotten my hands on a Windows machine and been investigating this issue further. I've tested my solution on both Chrome and Firefox with NVDA.

- It seems EMPTY (or BLANK as it was announced on my machine) is exclusive to Firefox (doesn't occur on Chrome)
- BLANK is announced when the `input` is focused, to announce that the input is empty.
- This doesn't seem to be a bug (digging through NVDA issues) and falls in line with how NVDA announces inputs.

The BLANK announcement occurs for a NgSelect if it has an input. BLANK refers to the input not having a value. I can't think of a proper fix to omit "BLANK" from being announced.

The only way to satisfy "BLANK" would be to populate the input with the actual value. I don't believe this is compatible with how NgSelect presents values. 

 I don't think there is anything we can do to satisfy the announcement in this case. Though I would love to be proven wrong.
